adapters/repos/db/search.go

```diff
@@ -176,6 +179,71 @@ func hydrateObjectsIntoSearchResults(objs []*storobj.Object, dists []float32) se
 	return res
 }
 
+func (d *DB) ReferenceVectorSearch(ctx context.Context,
```

Comment:
Two questions:

1. Does this method (and it's children) really need to be in the `db` package. At first glace, it looks like it only uses existing public methods of the `db.Repo` type and could therefore easily be elevated to be a usecase level? 

2. Can you add a Godoc comment to describe what it does and anything else that the caller should know? [Example from another package](https://pkg.go.dev/github.com/semi-technologies/weaviate@v1.15.1/adapters/repos/db/lsmkv#Bucket.Delete).

Reply:
This method was moved into the ref2vec-centroid module